packages/langium/src/services.ts

```diff
@@ -75,6 +76,15 @@ export type LangiumLspServices = {
     Formatter?: Formatter
 }
 
+export interface InitializableService<T extends object> {
+    /**
+     * Initializes the service with client (IDE) related capabilities.
+     *
+     * This method is only executed if Langium is running as part of a language server.
+     */
+    initialize?(clientCapabilities?: T): MaybePromise<void>
```

Comment:
A naive question: When there is setup code, do we also need teardown code for removing the created infrastructure or whatever was used? Then this interface could be a life cycle service.

Reply:
We can maybe add something like this later, but for now there's no need for it. We don't really "setup" anything here, but rather notify the LSP service of any existing client capabilities. The implementation is usually quite simplistic:

https://github.com/langium/langium/blob/245629e79008aea422c0cde36d0cb018453fbd2c/packages/langium/src/lsp/semantic-token-provider.ts#L163-L165

Reply:
It feels like a code smell to have this optional `initialize` method everywhere. Would it make sense to approach this with a life cycle event, like subscribing to the event when client capabilities are ready and then pick the information out of it?